pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java

```diff
@@ -110,6 +110,12 @@ public QuotaCheckerResponse isSegmentStorageWithinQuota(@Nonnull File segmentFil
       throw e;
     }
 
+    if (tableSubtypeSize.estimatedSizeInBytes == -1) {
```

Comment:
will this result in rejecting the segment?

Reply:
Yes, thats true. I haven't finalized this change yet (pending todos to say the least). The main point though is that estimate can be -1 and we shouldn't directly use it in the code that follows. If we go with the approach of caching, we could fall back to the last cached value etc; or accept the segment, but log a message/fire a metric that says we are flying blind a bit.

I can pull this out from this PR and leave only the logging in; we can review table-size estimations and address this issue along with others we would like to improve on.

Reply:
Based on offline discussion, we'll not reject segment in this case. I've made the handling of this case clearer.